### PR TITLE
Military Overhaul: the Terrain Combat Modifiers

### PR DESCRIPTION
**Pull request by zumbak04**
Changelog:
- Adapted terrain combat modifiers following pattern from https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/pull/744 :
  - Casters replace vanilla Pikemen.
  - Infantry get average modifier of vanilla Light Infantry and Heavy Infantry.
- Aerial Infantry get no malus from river crossings or landings
- Swarm get bonus fighting in deserts and jungles.

### CODE REVIEW

**Comment by MemeWizard** [comment]
Some numbers are just awfully precise
Can't you round them?

<details>
<summary>Click to expand</summary>

image

</details>

File: common/static_modifiers.txt [hunk 1]
```diff
@@ -279,142 +279,411 @@ settled_adventurer = {
...
-	heavy_infantry_defensive = 0.2
-	horse_archers_defensive = 0.15
-	light_cavalry_defensive = 0.15
-
-	archers_offensive = 0.2
+	large_cavalry_defensive = 0.2
+	war_machines_defensive = 0.2
 	pikemen_offensive = 0.1
+	large_cavalry_offensive = 0.1
+	war_machines_offensive = 0.1
```
Comment (MemeWizard) [inline common/static_modifiers.txt]: But it's not light_infantry ?

File: common/static_modifiers.txt [hunk 2]
```diff
@@ -279,142 +279,411 @@ settled_adventurer = {
...
 }
 
 defenderbonus_jungle = {
 	icon = 1
-	archers_defensive = 0.3
-	light_infantry_defensive = 0.5
-	pikemen_defensive = 0.2
-	heavy_infantry_defensive = 0.1
+	
+	# light_infantry $ replace pikemen
```
Comment (MemeWizard) [inline common/static_modifiers.txt]: $

**Comment by zumbak04** [comment]
> Some numbers are just awfully precise
> Can't you round them?

Round to what?

File: common/static_modifiers.txt (hunk 1 above)
Comment (zumbak04) [reply common/static_modifiers.txt]: Fixed!

File: common/static_modifiers.txt (hunk 2 above)
Comment (zumbak04) [reply common/static_modifiers.txt]: Fixed!

**Comment by MemeWizard** [comment]
> round to what?

To less eye-hurting numbers?

**Comment by zumbak04** [comment]
> To less eye-hurting numbers?

REEE! What number?

**Comment by MemeWizard** [comment]
> REEE! 

no u


> What number?

Like. all in strait_crossing and amphibious_landing

**Comment by zumbak04** [comment]
> Like, all in strait_crossing and amphibious_landing

Rounded to 0.025. I hope it works.

**Pull request merged**